pkg/cnb/remote_store_reader.go

```diff
@@ -22,9 +22,15 @@ func (r *RemoteStoreReader) Read(storeImages []v1alpha1.StoreImage) ([]v1alpha1.
 
 	c := make(chan v1alpha1.StoreBuildpack)
 	for _, storeImage := range storeImages {
-		storeImage := storeImage
+		storeImageCopy := storeImage
 		g.Go(func() error {
-			image, _, err := r.RegistryClient.Fetch(r.Keychain, storeImage.Image)
+			image, _, err := r.RegistryClient.Fetch(r.Keychain, storeImageCopy.Image)
+			if err != nil {
+				return err
+			}
+
+			bpMetadata := BuildpackageMetadata{}
+			err = imagehelpers.GetLabel(image, buildpackageMetadataLabel, &bpMetadata)
```

Comment:
What happens if this is builder image without the label?

I know the capi team is still planning on doing that.

Reply:
I did not realize we'd have to handle that case. I suppose we could leave the buildpackage info blank. Thats the best I can think of, considering those are non-conformant buildpackages, in my opinion.

Reply:
It might be good idea to move to only supporting buildpackages.

Reply:
Actually, they might just have been doing that just to get started because at the time there were not really any real buildpackages.

Reply:
I think we can safely ignore that case.